### PR TITLE
58 fix fetch memo logic

### PR DESCRIPTION
**Pull request by Leepak777**
createMemo n longer update the memos list for user
updated Dashboard, Savedlocation, and lens  to fetch memos list from memo controller get_all_memos

### CODE REVIEW

File: frontend/src/components/Dashboard/page.tsx [hunk 1]
```diff
@@ -14,45 +13,28 @@ const Dashboard = () => {
   const [selectedLocationPop, setSelectedLocationPop] = useState(null);
   const [expandedLocations, setExpandedLocations] = useState({});
   const [showPopup, setShowPopup] = useState(false);
-  const [memos, setMemos] = useState({});
-  const [reloadDashboard, setReloadDashboard] = useState(true);
+  const [memos, setMemos] = useState<any[]>([]);
```
Comment (jerrydngzh) [inline frontend/src/components/Dashboard/page.tsx:16]: we know the type of memos being fetched, why is this `any`?

File: frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx [hunk 2]
```diff
@@ -1,41 +1,29 @@
 //@ts-nocheck
```
Comment (jerrydngzh) [inline frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx:1]: remove this an all other mentions of `//@ts-nocheck` in the files you've changed

File: frontend/src/components/Dashboard/page.tsx [hunk 3]
```diff
@@ -14,45 +13,28 @@ const Dashboard = () => {
...
+      });
+  
       setLocations((prevLocations) => ({
         ...prevLocations,
         ...fetchedLocations,
       }));
+
     } catch (error) {
-      console.error("Error fetching memos:", error);
+      console.error("Error fetching data:", error);
```
Comment (jerrydngzh) [inline frontend/src/components/Dashboard/page.tsx:35]: properly catch the errors here. This is for things like 5XX, etc. 

If the error was a 404 it means that there isn't any error, just that the user doesn't have any memos so you can silently "fail".

File: frontend/src/components/Lens/page.tsx
Comment (jerrydngzh) [inline frontend/src/components/Lens/page.tsx:1]: Why did you change this? Lens isn't part of this issue for resolving memo fetch logic on dashboard/create memo.

@vinn03 has changes in his branch (#36) in this file, where he might've done refactor on, so this already has conflicts

**Review by jerrydngzh — CHANGES_REQUESTED** [review]

File: frontend/src/components/Lens/page.tsx
Comment (Leepak777) [reply frontend/src/components/Lens/page.tsx:1]: Lens also iterates through the user memos list to get all memos, I will roll back the commit and update when his changes are done

File: frontend/src/components/Dashboard/page.tsx (hunk 1 above)
Comment (Leepak777) [reply frontend/src/components/Dashboard/page.tsx:16]: originally I use MemoType but it doesn't save the memo._id
handleDelete 
setMemos(prevMemos => prevMemos.filter(m => m._id !== memo._id));
won't be able to use m._id to compare with the delete memo_id

File: frontend/src/components/Dashboard/page.tsx (hunk 3 above)
Comment (Leepak777) [reply frontend/src/components/Dashboard/page.tsx:35]: so something like 
catch (error) {
      if (error.response && error.response.status >= 500 && error.response.status < 600) {
        console.error("Server Error:", error);
      } 
    }
and ignore 404?

File: frontend/src/components/Dashboard/page.tsx (hunk 1 above)
Comment (jerrydngzh) [reply frontend/src/components/Dashboard/page.tsx:16]: I'm not sure what you mean by this. 

`MemoType`s `id` field is optionally null, if that was the error you were facing, then you can apply `!` after it to force the TS compiler that it isn't null. 

Otherwise, when you are at the stage to delete a memo, the memo id should exist.

File: frontend/src/components/Dashboard/page.tsx (hunk 3 above)
Comment (jerrydngzh) [reply frontend/src/components/Dashboard/page.tsx:35]: just have a case for 404 to return

File: frontend/src/components/Dashboard/page.tsx (hunk 1 above)
Comment (Leepak777) [reply frontend/src/components/Dashboard/page.tsx:16]: so when the memo was created as MemoType, it has no id since id is generated after the post request.
That's why here when I fetch the memo as MemoType, they have no memo.id, so instead of making it MemoType I made it any so that i can access memo._id

Comment (Leepak777) [reply frontend/src/components/Dashboard/page.tsx:16]: I sorted it out, const [memos, setMemos] = useState<MemoType[]>([]);

**Review by jerrydngzh — APPROVED** [review]

**Comment by jerrydngzh** [comment]
#36 has been merged into main, so you can either setup those changes here or in a different branch, otherwise lgtm

**Review by jerrydngzh — APPROVED** [review]
lgtm, just did some refactor ontop
